transport/conn.go

```diff
@@ -171,6 +183,7 @@ func (c *connReader) handler(streamID frame.StreamID) ResponseHandler {
 	return h
 }
 
+// loop terminates when its connection gets closed by the pool, especially when session context is done.
```

Comment:
I don't see any context being used here, why there's a hidden side effect?

Reply:
The way reader loop works is that its blocking on io.Copy which from what I recall cannot be interrupted using contexts.

When session is being closed it closes all its connections, making io.Copy return an io error, closing the reader loop.
This occurs especially when session context is closed.